### DISCUSSION TITLE
Run without keyboard

### DISCUSSION THREAD

**Discussion by dhajoglou** [raysan5/raylib#2354] [Q&A]
I'm making a remote appliance that's controlled by a REST api and used w/o a keyboard. I intend to start the program on startup with either systemd or rc.local which I'd normally start as a background process.  It's your basic remote display concept (though, in this case, it's generating lower thirds for a race timing system and plugged into a video mixer/switcher).

However, if I attempt to background, for example, core_basic_window, it just hangs on initWindow and will resume if I send it back to the foreground. Is there a way to tell raylib to forget about the keyboard and just move on?

I'm using a Raspberry Pi 3 (buster) and raylib 4.0.0.

**Comment by dhajoglou** [comment]
Ok, so I got it partially wrong.  I still can't run/background a task on the command line but I can put it into /etc/rc.local with a background and on boot the app will start and produce graphics.

Still, is there a way to start/background a raylib linked process from the command line?

**Comment by raysan5** [comment]
@dhajoglou I'm not sure if I understand the question... You can `SetConfigFlags(FLAG_WINDOW_ALWAYS_RUN)` before `InitWindow()` to keep the process running when minimized. You can also enable `SUPPORT_CUSTOM_FRAME_CONTROL` on `config.h` and recompile raylib to support [custom frame control](https://github.com/raysan5/raylib/blob/master/examples/core/core_custom_frame_control.c).

Not sure if those solutions could help for your use case...

**Reply by dhajoglou** [reply]
I found that if I did ./core_basic_window & it would just block in InitWindow.  If, however, I used nohup (presumably because of the non-tty) it happily ran in the background. Running from rc.local with & also works like nohup.  I'll give the flag method a try though I am happy that nohup works. I'm working on a system that I'd like to be able to manage remotely w/o leaving the ssh session open after starting.

thanks for your comments!

**Reply by reinhardkrakow** [reply]
Hi,
I had the same problem (running my program in background).

if I run my program with the command ./myProgram & , RPi did not give me the prompt after and nothing happens on the screen. On the console, if I hit <RETURN> the program stops.

I resolve the problem by using my own custom logger like showed in the file myProgram.c with comment all the lines with print output. 
It was still not possible to run my program with the command ./myProgram &. The program looks like it was running because after starting the program the RPi, I receive the PID of the program and the prompt.

So I decide to run my program as a service. I create the file /etc/systemd/system/myProgram.service showed in the file myProgram.service.
I this file, I remove the '&' after the command.

With the command sudo systemctl enable myProgram.service and the command sudo systemctl start myProgram.service, myProgram is running on background. After reboot, myProgram was running automatically.
[myProgram.zip](https://github.com/raysan5/raylib/files/8662011/myProgram.zip)

BR
Reinhard
